### PR TITLE
Add documentation for the satellite communication system

### PR DESCRIPTION
**Pull request by acfloria**
The documentation covers some general aspects of the satellite communication system, how to set it up and how to use it. In additional some troubleshooting inputs are given.

### CODE REVIEW

File: en/advanced_features/satcom.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
+# Satellite Communication System Manual
+The satellite communication system ensures a long range high latency link between the ground station and the vehicle using RockBlock as the service provider for the Iridium SBD Satellite Communication System. Given good signal quality the latency is between 10 to 15 seconds.
+
+ The link running cost consists of a line rental and per message cost:
+* Each module needs to be activated which costs £10.00 per month
+* Each message transmitted over the system costs one Credit per 50 bytes. Bundles of Credits can be bought from RockBlock for £0.04-£0.11 per Credit depending on the bundle size.
+
+The following components are required for an operating satellite commuication link:
```
Comment (hamishwillee) [inline en/advanced_features/satcom.md]: Spelling: commuication

**Comment by hamishwillee** [comment]
@acfloria Needs a subedit (I can do after merging) but looks good to me. See notes inline.

@bkueng Other than doc restructuring, is there anything you think is missing to use this?

File: en/advanced_features/satcom.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+* A [RockBlock 9603](http://www.rock7mobile.com/products-rockblock-9603) module connected to a Pixhawk flashed with the PX4 Autopilot.
+* A message relay server running Ubuntu.
+* A groundStation computer with QGroundControl installed and running Ubuntu as the operating system.
```
[the remark this answers is not in the document]
Comment (acfloria) [reply en/advanced_features/satcom.md]: fixed

File: en/advanced_features/satcom.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+Notes: 
+* It might be possible to run the system also on other OS but this has not been tested yet and therefore is not guaranteed to work.
```
[the remark this answers is not in the document]
Comment (acfloria) [reply en/advanced_features/satcom.md]: Changed

File: en/advanced_features/satcom.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
+# Satellite Communication System Manual
+The satellite communication system ensures a long range high latency link between the ground station and the vehicle using RockBlock as the service provider for the Iridium SBD Satellite Communication System. Given good signal quality the latency is between 10 to 15 seconds.
+
+ The link running cost consists of a line rental and per message cost:
+* Each module needs to be activated which costs £10.00 per month
+* Each message transmitted over the system costs one Credit per 50 bytes. Bundles of Credits can be bought from RockBlock for £0.04-£0.11 per Credit depending on the bundle size.
+
+The following components are required for an operating satellite commuication link:
+* A [RockBlock 9603](http://www.rock7mobile.com/products-rockblock-9603) module connected to a Pixhawk flashed with the PX4 Autopilot.
+* A message relay server running Ubuntu.
```
[the remark this answers is not in the document]
Comment (acfloria) [reply en/advanced_features/satcom.md]: It should not but just to be safe I added the versions we tested.

File: en/advanced_features/satcom.md [hunk 5]
```diff
@@ -0,0 +1,218 @@
+# Satellite Communication System Manual
+The satellite communication system ensures a long range high latency link between the ground station and the vehicle using RockBlock as the service provider for the Iridium SBD Satellite Communication System. Given good signal quality the latency is between 10 to 15 seconds.
```
[the remark this answers is not in the document]
Comment (acfloria) [reply en/advanced_features/satcom.md]: Ok, changed

File: en/advanced_features/satcom.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
+# Satellite Communication System Manual
```
[the remark this answers is not in the document]
Comment (acfloria) [reply en/advanced_features/satcom.md]: I agree and changed to your suggestion

File: en/advanced_features/satcom.md (hunk 1 above)
Comment (acfloria) [reply en/advanced_features/satcom.md]: Fixed

File: en/advanced_features/satcom.md [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+![Architecture](../../assets/satcom/architecture.jpg)
+
+## Set-up
+
+In this section the set-up of the individual system components is described.
+
+#### Vehicle
+###### Wiring
+Connect the RockBlock module to a serial port of the Pixhawk. Due to the power requirements of the module it can only be powered over a high-power serial port as a maximum of 0.5 A at 5 V are required. If none is available/free then another power source which has the same ground level as the Pixhawk and can provide required power has to be setup. The details of the [connectors](https://docs.rockblock.rock7.com/docs/connectors) and the [power requirements](https://docs.rockblock.rock7.com/docs/power-supply) can be found in the RockBlock documentation.
```
Comment (bkueng) [inline en/advanced_features/satcom.md]: Do we have documentation which UARTs are high-power?

File: en/advanced_features/satcom.md [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+It is expected that the relay server is either using the Ubuntu 16.04 or 14.04 OS. 
+* The server working as a message relay should have a static IP address and two publicly accessible, open TCP ports:
+
+    *  `5672` for the RabbitMQ message broker (can be changed in the rabbitmq settings)
+    * `45679` for the HTTP POST interface (can be changed in the `relay.cfg` file)
+
+* Install the required python modules:
+
+```
+    sudo pip install pika
```
Comment (bkueng) [inline en/advanced_features/satcom.md]: Reduce to one line: `sudo pip install pika tornado future`?

File: en/advanced_features/satcom.md (hunk 7 above)
Comment (acfloria) [reply en/advanced_features/satcom.md]: The only info I found was in the [ardupilot telemetry doc](http://ardupilot.org/copter/docs/common-telemetry-port-setup-for-apm-px4-and-pixhawk.html) but I am not sure if that is true for all boards. In general I also think that Telem1 is already used for the regular telemetry.

File: en/advanced_features/satcom.md (hunk 8 above)
Comment (acfloria) [reply en/advanced_features/satcom.md]: Ok, changed.

**Comment by hamishwillee** [comment]
@acfloria Thanks for the changes. Let me know when you're done (ie discussion with Beat is complete) and I'll merge/further subedit if needed.

File: en/advanced_features/satcom.md (hunk 7 above)
Comment (bkueng) [reply en/advanced_features/satcom.md]: Hmm I don't know either. The APM docs only mention Telem1. Do you use it on that port?
@davids5 do you know about the maximum Amps for UART ports on Pixhawks?

Comment (acfloria) [reply en/advanced_features/satcom.md]: No, we are using the Serial6 and power it from an external source.

Comment (davids5) [reply en/advanced_features/satcom.md]: @bkueng Clasicaly TELEM1 has the "High Power" supply.

FMUv2- BQ24315 Powers SERIAL1 (TELEM1) as VDD_5V_HIPOWER  1.5A max. It is controlled and has over current sensing.

FMUv4 - has a 1 A fuse. It is not controlled and has no over current sensing.

FMUv4pro - BQ24315 Powers TELEM1 as VDD_5V_HIPOWER  1.0A max.  It is controlled and has over current sensing.

FMUv5 HW - BQ24315 Powers TELEM1 as VDD_5V_HIPOWER  1.5A max.  It is controlled and has over current sensing.

**Comment by hamishwillee** [comment]
@acfloria @bkueng I have merged this, and subedited (including renaming doc) in https://github.com/PX4/px4_user_guide/pull/311 (also merged). Looked pretty good before, but I think even easier to read now. Florian, in particular, the architecture diagram moved up a tiny bit - it is really useful.

Things that still need to be discussed:
1. IMO it is  worth adding David's information about power connections/UARTs to use for different Pixhawks. Leaving to you
1. This uses https://github.com/acfloria/SatComInfrastructure.git on @acfloria repo. Should this be owned as part of dronecode or PX4 org on github?

**Comment by bkueng** [comment]
> IMO it is worth adding David's information about power connections/UARTs to use for different Pixhawks. Leaving to you

Definitely worth adding. Should we add it to the individual board pages (e.g. TELEM1 pinout)? Something like:

Pin | Signal | Volt
--- | --- | ---
1 (red) | VCC       | +5V *
2 (blk) | TX (OUT)  | +3.3V
3 (blk) | RX (IN)   | +3.3V
4 (blk) | CTS (IN)  | +3.3V
5 (blk) | RTS (OUT) | +3.3V
6 (blk) | GND       | GND

\* 1.5A max. It is controlled and has over current sensing.

**Comment by hamishwillee** [comment]
>  Should we add it to the individual board pages (e.g. TELEM1 pinout)? 

Yes. 

We might say what this is normally intended to power.

Normally we only talk about power in context of Pixhawk not being able to power servo rails. Feels odd to see that the Pixhawk can power anything.

**Comment by acfloria** [comment]
For me it would be fine if the SatComInfrastructure Repo would be owned by PX4 or Dronecode.

**Comment by hamishwillee** [comment]
@bkueng Up to you to discuss repo ownership with @acfloria . Cheers

**Pull request merged**